cabal-install/Distribution/Client/Status.hs

```diff
@@ -0,0 +1,353 @@
+      return $ (Just sandboxPath, Just $ sandboxPackageDBPath sandboxPath comp platform)
+    else return (Nothing, Nothing)
+  let dbs :: [PackageDB]
+      dbs = GlobalPackageDB
+          : UserPackageDB
+          : case sandboxDbPath of
+              Nothing -> []
+              Just p -> [SpecificPackageDB p]
+  case () of { () -> do
+    printCurrentVersion
+    printCabalDirVersion
+    printProgramVersions
+    printPackageInformation
+    printSandboxInformation
+    printPackageDbs
+    printPackageDBChecks
+    -- printInstallPlan is called by printPackageInformation
+   where
+  printCurrentVersion :: IO ()
+  printProgramVersions :: IO ()
+  printCabalDirVersion :: IO ()
+  printPackageInformation :: IO ()
+  printPackageDBChecks :: IO ()
+  printSandboxInformation :: IO ()
+  printPackageDbs :: IO ()
+  printInstallPlan :: IO ()
+  printCurrentVersion = putStrLn
+                      $    "cabal-install version "
+                        ++ display Paths_cabal_install.version
+                        ++ " compiled using version "
+                        ++ display cabalVersion
+                        ++ " of the Cabal library "
+  printCabalDirVersion = do
+    defInstDirs <- LBI.defaultInstallDirs defaultCompiler True False
+    let binDir :: FilePath
+        binDir = LBI.fromPathTemplate
+               $ LBI.bindir
+               $ LBI.substituteInstallDirTemplates [] defInstDirs
+    currentBinDir <- getExecutablePath
+    when (takeDirectory binDir /= takeDirectory currentBinDir) $ do
+      let defaultCabalPath = binDir `combine` ("cabal" <.> exeExtension)
+      exists <- doesFileExist defaultCabalPath
+      when exists $ do
+        -- TODO: replace rawSystemStdout by rawSystemStdInOuts
+        v <- rawSystemStdout verbosity defaultCabalPath ["--version"]
+        let vs :: [Maybe Version]
+            vs = map (simpleParse . filter (\x -> x `elem` "0123456789.")) $ lines v
+        case vs of
+          [Just cabalInstallVersion, Just _cabalVersion] ->
+            if cabalInstallVersion == Paths_cabal_install.version
+              then return ()
+              else putStrLn $ "There is a different cabal (version "
+                            ++ showVersion cabalInstallVersion
+                            ++ ") located in " ++ binDir
+          _ -> putStrLn $ "Warning: could not parse version of the `cabal`"
+                       ++ " executable located in " ++ binDir
+    return ()
+  printProgramVersions = do
+    when (Cabal.fromFlag (statusProgVersions statusFlags)
+       || Cabal.fromFlag (statusAll statusFlags)) $ do
+      putStrLn "program versions:"
+      flip mapM_ (configuredPrograms conf) $ \cp ->
+        case programVersion cp of
+          Nothing -> return ()
+          Just v  -> putStrLn
+                   $ render
+                   $ nest 2
+                   $    text (programId cp)
+                     <> text "-"
+                     <> disp v
+    putStrLn "configured compiler:"
```

Comment:
I think we should use proper punctuation here (and elsewhere). That is, `s/configured compiler/Configured compiler:/`.